src/main/kotlin/graphics/scenery/controls/behaviours/VRTouch.kt

```diff
@@ -121,14 +102,47 @@ open class VRTouch(
                                 controller.children.first(),
                                 device,
                                 { scene.discover(scene, { n -> n.getAttributeOrNull(Touchable::class.java) != null }) },
-                                if (vibrate) fun(){ (hmd as? OpenVRHMD)?.vibrate(device) } else fun(){})
+                                if (vibrate) fun(){ (hmd as? OpenVRHMD)?.vibrate(device); onTouch?.invoke() } else onTouch)
                             future.complete(touchBehaviour)
                         }
                     }
                 }
             }
             return future
         }
+
+        /**
+         * Apply the [Touchable.onHoldChangeDiffuseTo] color.
+         * If you are calling this manually make sure [unapplySelectionColor] will be called later.
+         */
+        fun applySelectionColor(node: Node) {
+            val touchable = node.getAttributeOrNull(Touchable::class.java)
+            val material = node.materialOrNull()
+
+            // if the following is set, some other VRTouch is already touching this
+            // and we don't want to interfere
+            if (touchable?.originalDiffuse == null
+                && touchable?.onHoldChangeDiffuseTo != null
+                && material != null
+            ) {
+                touchable.originalDiffuse = material.diffuse
+                material.diffuse = touchable.onHoldChangeDiffuseTo
+            }
+        }
+
+        /**
+         * Return to the original diffuse color after calling  [applySelectionColor].
+         * Should do nothing if no previous call of [applySelectionColor] happend.
+         */
+        fun unapplySelectionColor (node: Node) {
```

Comment:
Unwanted space after function name. Yes, I know, style file is coming at some point ^^